app/src/main/java/lt/vilnius/tvarkau/fragments/ProblemsMapFragment.java

```diff
@@ -1,14 +1,155 @@
 package lt.vilnius.tvarkau.fragments;
 
+import android.Manifest;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.os.Bundle;
+import android.support.v4.content.ContextCompat;
+
+import com.google.android.gms.maps.CameraUpdateFactory;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.OnMapReadyCallback;
 import com.google.android.gms.maps.SupportMapFragment;
+import com.google.android.gms.maps.model.BitmapDescriptor;
+import com.google.android.gms.maps.model.BitmapDescriptorFactory;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.Marker;
+import com.google.android.gms.maps.model.MarkerOptions;
+
+import java.util.HashMap;
+
+import lt.vilnius.tvarkau.ProblemDetailActivity;
+import lt.vilnius.tvarkau.R;
+import lt.vilnius.tvarkau.entity.Problem;
+import lt.vilnius.tvarkau.factory.DummyProblems;
+import lt.vilnius.tvarkau.utils.PermissionUtils;
+import lt.vilnius.tvarkau.views.adapters.MapsInfoWindowAdapter;
 
 /**
  * Created by Karolis Vycius on 2016-01-13.
  */
-public class ProblemsMapFragment extends SupportMapFragment {
+public class ProblemsMapFragment extends SupportMapFragment
+        implements OnMapReadyCallback,
+        GoogleMap.OnInfoWindowClickListener,
+        GoogleMap.OnInfoWindowCloseListener,
+        GoogleMap.OnMarkerClickListener {
+
+    protected static final LatLng VILNIUS_LAT_LNG = new LatLng(54.687157, 25.279652);
+    protected static final int GPS_PERMISSION_REQUEST_CODE = 11;
+
+    protected GoogleMap googleMap;
+
+    protected BitmapDescriptor inProgressMarker;
+    protected BitmapDescriptor doneMarker;
+
+    protected HashMap<String, Problem> problemHashMap = new HashMap<>();
 
     public static ProblemsMapFragment getInstance() {
         return new ProblemsMapFragment();
     }
 
+    @Override
+    public void onActivityCreated(Bundle savedInstanceState) {
+        super.onActivityCreated(savedInstanceState);
+
+        setMarkerResources();
+
+        getMapAsync(this);
+    }
+
+    private void setMarkerResources() {
+        inProgressMarker = BitmapDescriptorFactory.fromResource(R.drawable.ic_pin_drop_blue);
+        doneMarker = BitmapDescriptorFactory.fromResource(R.drawable.ic_pin_drop_green);
+    }
+
+    protected void requestGPSPermission() {
```

Comment:
Modified `PermissionUtils` methods. Now it has only ability to check only if all required permissions are granted. Remove `try/catch` from `googleMap.setMyLocationEnabled(true);`. It shows lint warnings, but IMHO it's OK, because there is check inside `PermissionUtils`.